rainier-tests/src/test/scala/com/stripe/rainier/compute/RealTest.scala

```diff
@@ -129,7 +139,15 @@ class RealTest extends FunSuite {
     Real.gt(x, 0, x * x, x + 1)
   }
 
-  run("normal") { x =>
+  run("normal", defined = !_.isPosInfinity) { x =>
+    // FIXME: letting this be defined at +infinity results in the following error:
+    //     -Infinity did not equal NaN [c/ev, n=Infinity] (RealTest.scala:67)
+    // I.e. two different results from constant folding and evaluation.
+    // same applies to the "normal sum" test case below
+    Normal(x, 1).logDensity(Real(1d))
```

Comment:
This is a potential problem identified by the additional tests where constant folding and evaluation yield different results.
This is really a edge case but might a symptom of a more general problem.

The problematic case is excluded from the tests since I won't attempt to fix it in this PR but it probably warrants a separate bug report.